packages/runtime/test-runtime-utils/api-report/test-runtime-utils.api.md

```diff
@@ -476,6 +477,9 @@ export class MockFluidDataStoreRuntime extends EventEmitter implements IFluidDat
     readonly connected = true;
     // (undocumented)
     containerRuntime?: MockContainerRuntime;
+    create<T>(factory: {
```

Comment:
I'm not a huge fan of adding another `create` method specifically for tests. IMO our goal should be to make our tests align more with our production APIs and how they should be used, and to accomplish that we should instead do something like

- make `MockContainerRuntimeFactory` take a `ISharedObjectKind` in its constructor
- implement `createChannel` by asserting the `type` argument matches the type of the `ISharedObjectKind` factory passed into the constructor, then calling `.getFactory().create(..)`

This is a more reasonable approximation of what's done in the prod codepath: channel creation goes through the data store based on the data store's registry.
That will allow people to create e.g. SharedMaps using `SharedMap.create(new MockFluidDataStoreRuntime(), "id1")` in unit tests, which matches the API they'd use in production (but with a real data store implementation).

To make migration from old usage to new one more incremental (there's probably a lot of `MockDataStoreRuntime` in tests), you could make this approach optional and leave some instances of direct factory usage intact. Up to you.

Reply:
This create method matches the production API IFluidContainer.create (same input and same output).

We can't actually take in ISharedObjectKind as doing so creates a cyclic package dependency error.

While I'd like to make the mocks work more like the production stuff, in this case that seems a bit more complex and verbose that we want (the point of the mocks is to be able to work with fewer parts of the system, ex have a runtime without a container)

Unless I'm missing something, I think your suggestion would require the test to replace:
```typescript
return new MockFluidDataStoreRuntime().create(kind, id);
```

with something more like
```typescript
	// make this able to take in a set of factories like this
	const containerFactory = new MockContainerRuntimeFactory({sharedObjects: [[kind.getFactory()]]});
	const dataStoreRuntime = new MockFluidDataStoreRuntime();
	// Modify constructor of MockContainerRuntime so it modifies its provided dataStoreRuntime to inject the factory into it here.
	const container = containerFactory.createContainerRuntime(dataStoreRuntime);
	return kind.create(dataStoreRuntime, id);
```

every time a DDS instance was desired.

Given that would require adding an API to MockFluidDataStoreRuntime so someone can side-load factories into it, it seems like 
```typescript
	const runtime = new MockFluidDataStoreRuntime();
	runtime.addFactory(kind);
	kind.create(runtime, id);
```
Or
```typescript
	return kind.create(new MockFluidDataStoreRuntime({sharedObjects: [[kind.getFactory()]]}), id);
```

would make more sense.

Thats still a lot less concise what I added, and requires adding significant logic to the mock. Wrapping that extra boilerplate up in a helper is problematic since you would want something like:
```typescript
export function sharedObjectWithMockContainer<T>(
	kind: ISharedObjectKind<T>,
	id?: string,
): T {
	return kind.create(new MockFluidDataStoreRuntime({sharedObjects: [[kind.getFactory()]]}), id);
}
```

but we can't put that in with the mocks since they can't reference ISharedObjectKind.

Maybe if we move ISharedObjectKind to a different package (datastore-definitions?), when we can actually have the above helper and that approach makes sense?

Reply:
> This create method matches the production API IFluidContainer.create (same input and same output).

Sure, but it's at a totally different layer. The interface this class is mocking already has a production API for doing what `.create<T>` does (`createChannel`), except its implementation does nothing. If we wanted to sort out the real data store interface to support this `.create<T>` API and resolve any problems around it (there probably aren't many, though I'd have to think for a bit to confirm), then that seems fine.

> Unless I'm missing something, I think your suggestion would require the test to replace:

Not really. The registry would be on the mock data store. So more like:

```typescript
	const mockRuntime= new MockFluidDataStoreRuntime({ registry: [[SharedString.getFactory()]] });
	const myString = SharedString.create(mockRuntime, "idFoo");
```

We wouldn't necessarily need a registry and associated complications in the mocks. Perhaps we say the mocks only support instantiating a single shared object or something like that. But the above sample functions very closely to how the production codepath works.

Reply:
done.